Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/contactdiary/ui/ContactDiaryTestFragment.kt

```diff
@@ -0,0 +1,56 @@
+package de.rki.coronawarnapp.test.contactdiary.ui
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentTestContactDiaryBinding
+import de.rki.coronawarnapp.test.menu.ui.TestMenuItem
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+@SuppressLint("SetTextI18n")
+class ContactDiaryTestFragment : Fragment(R.layout.fragment_test_contact_diary), AutoInject {
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val vm: ContactDiaryTestFragmentViewModel by cwaViewModels { viewModelFactory }
+
+    private val binding: FragmentTestContactDiaryBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        vm.locationVisits.observe2(this) {
```

Comment:
probably better to map the data directly in vm?

Reply:
While for a test fragment some shortcuts may be taken, in general, try to avoid direct calls to viewmodels that return data, i.e. no "getters".